drivers/display/ssd1322.c

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2024 Lukasz Hawrylko
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/sys/util.h"
+#define DT_DRV_COMPAT solomon_ssd1322
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ssd1322, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/kernel.h>
+
+#define SSD1322_SET_COLUMN_ADDR      0x15
+#define SSD1322_ENABLE_RAM_WRITE     0x5C
+#define SSD1322_SET_ROW_ADDR         0x75
+#define SSD1322_SET_REMAP            0xA0
+#define SSD1322_SET_START_LINE       0xA1
+#define SSD1322_SET_DISPLAY_OFFSET   0xA2
+#define SSD1322_BLANKING_ON          0xA4
+#define SSD1322_BLANKING_OFF         0xA6
+#define SSD1322_EXIT_PARTIAL         0xA9
+#define SSD1322_DISPLAY_OFF          0xAE
+#define SSD1322_DISPLAY_ON           0xAF
+#define SSD1322_SET_PHASE_LENGTH     0xB1
+#define SSD1322_SET_CLOCK_DIV        0xB3
+#define SSD1322_SET_GPIO             0xB5
+#define SSD1322_SET_SECOND_PRECHARGE 0xB6
+#define SSD1322_DEFAULT_GREYSCALE    0xB9
+#define SSD1322_SET_PRECHARGE        0xBB
+#define SSD1322_SET_VCOMH            0xBE
+#define SSD1322_SET_CONTRAST         0xC1
+#define SSD1322_SET_MUX_RATIO        0xCA
+#define SSD1322_COMMAND_LOCK         0xFD
+
+struct ssd1322_config {
+	const struct device *mipi_dev;
+	struct mipi_dbi_config dbi_config;
+	uint16_t height;
+	uint16_t width;
+	uint16_t column_offset;
+	uint8_t row_offset;
+	uint8_t start_line;
+	uint8_t mux_ratio;
+	bool remap_row_first;
+	bool remap_columns;
+	bool remap_rows;
+	bool remap_nibble;
+	bool remap_com_odd_even_split;
+	bool remap_com_dual;
+	uint8_t segments_per_pixel;
+	uint8_t *conversion_buf;
+	uint16_t conversion_buf_size;
+};
+
+static inline int ssd1322_write_command(const struct device *dev, uint8_t cmd, const uint8_t *buf,
+					size_t len)
+{
+	const struct ssd1322_config *config = dev->config;
+
+	return mipi_dbi_command_write(config->mipi_dev, &config->dbi_config, cmd, buf, len);
+}
+
+static int ssd1322_blanking_on(const struct device *dev)
+{
+	return ssd1322_write_command(dev, SSD1322_BLANKING_ON, NULL, 0);
+}
+
+static int ssd1322_blanking_off(const struct device *dev)
+{
+	return ssd1322_write_command(dev, SSD1322_BLANKING_OFF, NULL, 0);
+}
+
+/**
+ * The controller uses 4-bit grayscale format, so one pixel is represented by 4 bits.
+ * Zephyr's display API does not support this format, so this uses mono01, and converts each 1-bit
+ * pixel to 1111 or 0000.
+ *
+ * @param in_buf: pointer to input buffer in mono01 format. This value will bel updated to point to
+ * the first byte after the last converted pixel.
+ * @param pixel_count: pointer to the total number of pixels in in_buf to convert. The number of
```

Comment:
The compliance check fails with a typo error here. How do i address that?

Reply:
Name it buf_in and buf_out :-). Btw, please do not use doxygen style here.

Reply:
nice workaround ;-)